### PR TITLE
Delete XmonSimulator

### PR DESCRIPTION
**Pull request by dabacon**
This simulator has only delivered marginal improvement over cirq.Simulator(), and then only on some machines.  

Functionality about only supporting some gates will come with the addition of gateset to device.

### CODE REVIEW

**Comment by dabacon** [comment]
Fixes #1978

**Comment by mpharrigan** [comment]
RIP

File: cirq/devices/gate_set.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class GateSpec():
+
+    def __init__(self, constructor, params):
+        pass
+
+    def canonical
+
+class GateSet():
```
Comment (mpharrigan) [inline cirq/devices/gate_set.py]: :eyes:

Comment (dabacon) [reply cirq/devices/gate_set.py]: ha yeah bad merge.

**Pull request merged**